extensions/Oversight/migrateToRevDel.php

```diff
@@ -0,0 +1,280 @@
+<?php
+/**
+ * Migrate revisions hidden with the Oversight extension to revdel.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ * http://www.gnu.org/copyleft/gpl.html
+ *
+ * @file
+ * @ingroup Maintenance
+ */
+
+// Detect $IP
+$IP = getenv( 'MW_INSTALL_PATH' );
+if ( $IP === false ) {
+	$IP = __DIR__ . '/../..';
+}
+
+// Require base maintenance class
+require_once( "$IP/maintenance/Maintenance.php" );
+
+/**
+ * Migrate revisions hidden with the Oversight extension to revdel.
+ *
+ * @ingroup Maintenance
+ */
+class MigrateOversightRevisions extends Maintenance {
+	public function __construct() {
+		parent::__construct();
+		$this->setBatchSize( 20 );
+		$this->mDescription = "Migrate revisions hidden with the Oversight extension to revdel.";
+		$this->addOption( 'migrate', "Turn off dry-run mode and actually insert OS data into revision/archive table." );
+		$this->addOption( 'max', "Maximum number of revisions to migrate at a time.", true );
+	}
+
+	/**
+	 * Copied from SpecialOversight::getSelectFields so the script
+	 * can run without the extension being enabled
+	 *
+	 * @return array
+	 */
+	private function getSelectFields() {
+		return array( 'hidden_page as page_id',
+			'hidden_namespace as page_namespace',
+			'hidden_title as page_title',
+
+			'hidden_page as rev_page',
+			'hidden_comment as rev_comment',
+			'hidden_user as rev_user',
+			'hidden_user_text as rev_user_text',
+			'hidden_timestamp as rev_timestamp',
+			'hidden_minor_edit as rev_minor_edit',
+			'hidden_deleted as rev_deleted',
+			'hidden_rev_id as rev_id',
+			'hidden_text_id as rev_text_id',
+
+			'0 as rev_len',
+
+			'hidden_by_user',
+			'hidden_on_timestamp',
+			'hidden_reason',
+
+			'user_name',
+
+			'0 as page_is_new',
+			'0 as rc_id',
+			'1 as rc_patrolled',
+			'0 as rc_old_len',
+			'0 as rc_new_len',
+			'0 as rc_params',
+
+			'NULL AS rc_log_action',
+			'0 AS rc_deleted',
+			'0 AS rc_logid',
+			'NULL AS rc_log_type',
+			'NULL AS rev_parent_id'
+		);
+	}
+
+	public function execute() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this function "execute" is 23 which is greater than 20 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This function "execute" has 186 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)